reviewer/metrics/Metrics.cpp

```diff
@@ -18,20 +18,45 @@
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 //
 
-#include "metrics/Workflow.hh"
+#include "metrics/Metrics.hh"
 
 #include <algorithm>
 #include <fstream>
 #include <map>
 #include <sstream>
 #include <vector>
 
+#include "thirdparty/json/json.hpp"
+
 using graphtools::NodeId;
 using std::map;
 using std::ofstream;
 using std::string;
 using std::vector;
 
+using Genotype = vector<int>;
+
+static map<string, Genotype> getGenotypes(const LocusSpecification& locusSpec, const GraphPaths& hapPaths)
+{
+    map<string, Genotype> genotypes;
+    for (const auto& variantSpec : locusSpec.variantSpecs())
+    {
+        assert(variantSpec.nodes().size() == 1);
+        const auto strNode = variantSpec.nodes().front();
+
+        Genotype genotype;
+        for (int hapIndex = 0; hapIndex != hapPaths.size(); ++hapIndex)
```

Comment:
This could be simplified to range based for.

Reply:
Good point!